app/models/util/user.py

```diff
@@ -0,0 +1,14 @@
+from app import db
+
+from app.models import User
+
+def get_or_create_user(*, name: str, email: str) -> int:
+    session = db.get_session()
+    user = session.query(User).filter(User.email == email).one_or_none()
+    if user:
+        return user.id
+
+    session.add(User(name=name, email=email))
+    session.commit()
+    user = session.query(User).filter(User.email == email).one_or_none()
```

Comment:
If you store the user object the id field gets populated after the commit: https://docs.sqlalchemy.org/en/latest/orm/tutorial.html#adding-and-updating-objects

Reply:
Also, what about returning the user object, instead of just ID?

Reply:
Also makes sense. Not sure if I want to assume the data stays the same across these helper functions though... Or assume you have the user object

Reply:
you do have the user... you are either finding it or creating it so... yeah
Also... maybe split this in two? I'd like to know what operation just took place, i think? 
This would help to make the flow to guide new users to team selection/creation.

Reply:
Or just another method somewhere called has_team 🤷

Reply:
Not sure I follow. Like a get_user() that gets a user given either email or ID, and another that actually creates the user regardless? Hmmmmm... I'm starting to see what you're getting at